axelrod/result_set.py

```diff
@@ -45,6 +47,8 @@ def __init__(self, players, interactions, progress_bar=True):
             progress_bar : bool
                 Whether or not to create a progress bar which will be updated
         """
+        if game is not None:
+            self.game = game
```

Comment:
Cool re that antipattern. The alternative I see (if we don't like setting the class attribute) is in the init:

```
if game is None:
    self.game = Game
else:
   self.game = game
```

I can change that here and also in the tournament class.